cacerts/detect.go

```diff
@@ -31,40 +33,62 @@ const (
 	PlanEntryCACertsHelper = "ca-certificates-helper"
 )
 
-// Detect always passes and optionally provides ca-certificates. If there is a binding of
+// Detect always passes by default and optionally provides ca-certificates. If there is a binding of
 // type "ca-certificates" Detect also requires ca-certificates and provides an array of certificate paths in the
 // plan entry metadata.
+//
+// To prevent default detection, users can set the
+// BP_ENABLE_RUNTIME_CERT_BINDING environment variable to "false" at
+// build-time. This will disable the helper layer, and the buildpack will only
+// detect if there is no ca-certificates binding present at build-time.
 func (Detect) Detect(context libcnb.DetectContext) (libcnb.DetectResult, error) {
-	result := libcnb.DetectResult{
-		Pass: true,
-		Plans: []libcnb.BuildPlan{
-			{
-				Provides: []libcnb.BuildPlanProvide{
-					{Name: PlanEntryCACertsHelper},
-					{Name: PlanEntryCACerts},
-				},
-				Requires: []libcnb.BuildPlanRequire{
-					{Name: PlanEntryCACertsHelper},
-				},
-			},
-			{
-				Provides: []libcnb.BuildPlanProvide{
-					{Name: PlanEntryCACertsHelper},
-				},
-				Requires: []libcnb.BuildPlanRequire{
-					{Name: PlanEntryCACertsHelper},
-				},
-			},
-		},
+	provides := []libcnb.BuildPlanProvide{
+		{Name: PlanEntryCACerts},
 	}
+
+	requires := []libcnb.BuildPlanRequire{}
+
+	// If there are CA cert bindings at build time, require PlanEntryCACerts
 	paths := getsCertsFromBindings(context.Platform.Bindings)
 	if len(paths) > 0 {
-		result.Plans[0].Requires = append(result.Plans[0].Requires, libcnb.BuildPlanRequire{
+		requires = append(requires, libcnb.BuildPlanRequire{
 			Name: PlanEntryCACerts,
 			Metadata: map[string]interface{}{
 				"paths": paths,
 			},
 		})
 	}
+
+	result := libcnb.DetectResult{
+		Pass: true,
+		Plans: []libcnb.BuildPlan{
+			{
+				Provides: provides,
+				Requires: requires,
+			},
+		},
+	}
+
+	// If BP_ENABLE_RUNTIME_CERT_BINDING = false, do not enable helper layer.
+	enableCertBinding := os.Getenv("BP_ENABLE_RUNTIME_CERT_BINDING")
```

Comment:
We should do two things slightly different when checking `BP_ENABLE_RUNTIME_CERT_BINDING`
1. Use `strconv.ParseBool` so that values like `False` or `f` are also interpreted as `false`
2. Use an instance of `libpak.ConfigurationResolver` to get the value instead of using `os.Getenv` directly. This allows default values to be configured in `buildpack.toml`.

See paketo-buildpacks/native-image for an [example](https://github.com/paketo-buildpacks/native-image/blob/main/native/detect.go#L104-L114).

Reply:
Ah I was looking for an example. This is super helpful, thanks!!